cobalt/browser/application.cc

```diff
@@ -619,19 +619,83 @@ void AddCrashLogApplicationState(base::ApplicationState state) {
              << "required version, so not sending application state.";
 }
 
+void RecordLoaderAppTimeMetrics(
+    const StarboardExtensionLoaderAppMetricsApi* metrics_extension) {
+  int64_t elf_load_duration_us =
+      metrics_extension->GetElfLoadDurationMicroseconds();
+  int64_t elf_decompression_duration_us =
+      metrics_extension->GetElfDecompressionDurationMicroseconds();
+
+  if (elf_load_duration_us < 0) {
+    return;
+  }
+  if (elf_decompression_duration_us < 0) {
+    return;
+  }
+  if (elf_decompression_duration_us > elf_load_duration_us) {
+    // The decompression duration is considered to be contained within the ELF
+    // load duration, so this inequality is unexpected.
+    return;
+  }
+
+  int64_t unexplained_duration_us =
+      elf_load_duration_us - elf_decompression_duration_us;
+
+  base::UmaHistogramTimes(
+      "Cobalt.LoaderApp.ElfLoadDuration",
+      base::TimeDelta::FromMicroseconds(elf_load_duration_us));
+  base::UmaHistogramTimes(
+      "Cobalt.LoaderApp.ElfDecompressionDuration",
+      base::TimeDelta::FromMicroseconds(elf_decompression_duration_us));
+
+  // "Unexplained" just means means that we haven't yet attempted to break this
+  // chunk of time into its components. It's included here to give context for
+  // the other durations, as recommended by
+  // https://chromium.googlesource.com/chromium/src/+/lkgr/docs/speed/diagnostic_metrics.md#summation-diagnostics.
+  base::UmaHistogramTimes(
```

Comment:
Just verifying you want `UmaHistogramTimes` as your histgogram. It's for events between 1 ms up to 10 seconds (50 buckets).

Reply:
Yeah, it should work reasonably well for us as we expect these durations to be from a few milliseconds to a few hundred milliseconds.

We probably could reduce the number or buckets (or increase granularity) a bit by using `UmaHistogramCustomTimes()`, but I don't think it's necessary.